pkg/fqdn/rulegen_test.go

```diff
@@ -276,3 +276,45 @@ func (ds *FQDNTestSuite) TestRuleGenUpdatesOnReplace(c *C) {
 	c.Assert(len(rules[0].Egress), Equals, 1, Commentf("Incorrect number of generated egress rules for testCase with single cached ToFQDNs DNS entry"))
 	c.Assert(len(rules[0].Egress[0].ToCIDRSet), Equals, 1, Commentf("Generated CIDR count is not the same as ToFQDNs DNS entries in cache"))
 }
+
```

Comment:
Would it be possible to also test the out-of-order handling in a until test for daemon? I think we can create the actual bad situation with the queue.

Reply:
Is there a way we can have a field in `toFQDNs` rules that has them marked for deletion? This could be done independently of the EventQueue since the actual rule content wouldn't be modified (?)

Reply:
Do you mean the `EgressRule` in `api.Rule.Egress`? We can add it but I'm not sure if we have any private fields on there right now or if we can (since swagger make these types for us). I think you mentioned it elsewhere but we could make it public but it's unclear how long a rule would be reported in this state (since, if we could detect this race condition we wouldn't enqueue the replacement rule anyway).

Separately, how did you see the data moving about? One frustrating thing is that once an api.Rule is added to the policy repository a copy is made. This was the original reason I had to resort to the UUID label since that was the only identifier that would stick. RuleGen also happens to make a copy that it passes to PolicyAdd. We could change that (but risk introducing a race on the rule objects themselves if multiple updates are enqueued).

Reply:
Yeah, there are probably a lot of small details that make what I am suggesting much more difficult. This implementation is fine :)